src/cpr_data_access/models/search.py

```diff
@@ -43,6 +46,23 @@ def query_string_must_not_be_empty(cls, query_string):
             raise QueryError("query_string must not be empty")
         return query_string
 
+    @field_validator("family_ids", "document_ids")
+    def ids_must_fit_pattern(cls, ids):
+        """
+        Validate that the family and document ids are ids.
+
+        Example ids:
+            CCLW.document.i00000004.n0000
+            CCLW.family.i00000003.n0000
+            CCLW.executive.10014.4470
+            CCLW.family.10014.0
+        """
+        if ids:
+            for i in ids:
+                if len(i.split(".")) != 4:
+                    raise QueryError(f"id does not seem valid: {i}")
+        return ids
```

Comment:
This seems like a useful function to use in any API we build that accepts family or doc IDs as a parameter. Is there any more pattern matching it's worth doing here, while still making it flexible enough for adding new data and document types in future?

E.g. `id.split(".")[0].isupper()`; or something to do with the format of the final two sections of the ID